src/pybind/mgr/smb/fs.py

```diff
@@ -108,3 +110,83 @@ def resolve_exists(
                 raise NotADirectoryError(volpath)
         log.debug('Verified that %r exists in %r', volpath, volume)
         return volpath
+
+# SIMPLER VERSION OF CACHE
+
+
+class SimpleCachingCephFSPathResolver(CephFSPathResolver):
+    """
+    A subclass of CephFSPathResolver that adds caching to the resolve method.
+
+    This implementation uses a TTL (Time-To-Live) cache rather than an LRU (Least
+    Recently Used) cache. The TTL cache is preferred in this scenario because
+    the validity of cached paths can be time-sensitive, and we want to ensure that
+    paths are refreshed after a certain period regardless of access frequency.
+    """
+
+    def __init__(self, mgr: Module_T) -> None:
+        super().__init__(mgr)
+        # Initialize a TTL cache
+        self._cache = cachetools.TTLCache(maxsize=512, ttl=5)
+
+    def resolve(
+        self, volume: str, subvolumegroup: str, subvolume: str, path: str
+    ) -> str:
+        cache_key = (volume, subvolumegroup, subvolume, path)
```

Comment:
I wonder if we should include path in the cached data. Generally the lookup call of the subvolume path is what needs to be cached, and those will vary less often than the entire path (subvolume + path). The downside is obviously that we can't simply use the result of `resolve` from the base class.

I wonder if we ought to add a `resolve_subvolume(self, volume: str, subvolumegroup: str, subvolume: str) -> str` method to the base class and cache that instead of the `resolve` method. WDYT?

Reply:
I see, yeah then it makes sense to just cache subvolume path & add `resolve_subvolume` and cache that in subclass.